### PR TITLE
Add support for container registry as build args

### PR DESCRIPTION
**Pull request by wunzeco**
- Allows passing `CONTAINER_REGISTRY` as docker build args

### CODE REVIEW

**Comment by alanplatt** [comment]
I would like to understand why we are doing this here

**Comment by wunzeco** [comment]
Container registry for dependencies image specified in multistage Dockerfile for microservices is currently hard coded, which effectively restricts where a build can happen. i.e. only on one cloud provider. 
We want to be able to build on either ECR or ACR more so as we are in the process of migrating from ECR to ACR. 
Setting the container registry to pull from as a build arg gives us the flexibility around that issue.

File: docker_build.sh [hunk 1]
```diff
@@ -19,7 +20,12 @@ default_build () {
      GRADLE_TASKS="clean build publish"
   fi
 
+  BUILD_ARGS="--build-arg GRADLE_TASKS='${GRADLE_TASKS}' --build-arg VERSION=${VERSION}"
+  if [ ! -z $CONTAINER_REGISTRY ]; then
```
Comment (fatal-exception) [inline docker_build.sh]: could use -n ?

Comment (fatal-exception) [reply docker_build.sh]: maybe best-practice is 
```
if [[ -n "${CONTAINER_REGISTRY}" ]] ; ...
```

**Comment by wunzeco** [comment]
```
16:51 $ export BB=''

17:00 $ test -n $BB || echo 'Empty'

17:01 $ test ! -z $BB || echo 'Empty'
Empty
```

**Pull request merged**